.github/workflows/build.yaml

```diff
@@ -0,0 +1,46 @@
+name: Main Build
+
+on:
+    workflow_dispatch:
+    pull_request:
+        branches: [main]
+env:
+  GPG_PASSPHRASE: ${{secrets.GPG_PASSPHRASE}}
+  GPG_KEY: ${{secrets.GPG_KEY}}
+
+jobs:
+    build-push-wrapping-artefacts:
+        name: Building and Deploying Maven artefcats
+        runs-on: ubuntu-latest
+
+        steps:
+          - name: Checkin Code
+            uses: actions/checkout@v4
+          
+          - name: Set up JDK 11
+            uses: actions/setup-java@v4
+            with:
+              java-version: '11'
+              distribution: 'temurin'
+              server-id: 'github'
+              server-username: GITHUB_ACTOR
+              server-password: GITHUB_TOKEN
+              settings-path: '~/.m2'
+              gpg-private-key: ${{secrets.GPG_KEY}}
+              gpg-passphrase: GPG_PASSPHRASE
```

Comment:
shouldn't this be a variable?

'''yaml
  gpg-passphrase:
    description: 'Environment variable name for the GPG private key passphrase. Default is
       $GPG_PASSPHRASE.'
    required: false
'''